hardhat-tests/test/internal/hardhat-network/provider/modules/evm.ts

```diff
@@ -146,6 +146,21 @@ describe("Evm module", function () {
           assert.strictEqual(totalOffset1, originalOffset + offset1);
           assert.strictEqual(totalOffset2, originalOffset + offset1 + offset2);
         });
+
+        it("should provide a reasonable error for big numbers", async function () {
+          await assertInvalidArgumentsError(
+            this.provider,
+            "evm_increaseTime",
+            ["10000000000000000000000000000000"],
+            "Timestamp must be a positive integer or a string not exceeding 2^64 - 1"
+          );
+          await assertInvalidArgumentsError(
+            this.provider,
+            "evm_increaseTime",
+            ["0x1111111111111111111111111111111111111"],
+            "Timestamp must be a positive integer or a string not exceeding 2^64 - 1"
+          );
```

Comment:
Can we add 4 test cases more here?

1. Using exactly `2^64` produces the error too (using both hex-prefixed strings and decimal numbers)
2. Using exactly `2^64-1` doesn't produce the error (both hex-prefixed and decimal)

Reply:
I did that for the `evm_setNextBlockTimestamp`.

However, I now realized that `evm_increaseTime` actually internally converts the value to an `i64`... Does that mean we should support negative as well? This also means supporting up to 2**63-1. Is this something we want to pursue as part of this PR?

Reply:
IIRC, `evm_increaseTime` was allowed to receive negative values in the past, but this was a bug. We fixed that in EDR.

The reason that we use an `i64` in `ProviderData` is that the `fn block_time_offset_seconds` can return a negative value, if the first block is mined in the past.

Potentially we want to change that in the future and return an error. This might have been designed this way in Hardhat on purpose though.

I'd refer to @fvictorio for historic context.

Reply:
> Potentially we want to change that in the future and return an error.

No, this is not an error condition, it happens all the time. I don't remember exactly why; I think it's because in Hardhat the config is eagerly loaded (which then uses `now()` for the initial timestamp) but the node is created lazily after the first request, so normally the time offset will be `-N`, where `N` is the number of seconds that happened between the start of the task and the first request.

I'm not sure about the `i64` part, but I'm happy with just having those tests for `evm_setNextBlockTimestamp`, so no need to block on that.

Reply:
It looks the situation may not be obvious on the `i64` front, so I'll wait for @Wodann here. When everyone agrees, I think we could finally merge this.

Reply:
Approved.

Reply:
Thanks!